application/controllers/ErrorController.php

```diff
@@ -86,9 +86,20 @@ public function errorAction()
                         Logger::error("%s\n%s", $exception, $exception->getTraceAsString());
                         break;
                 }
-                $this->view->message = $exception->getMessage();
+
                 if ($this->getInvokeArg('displayExceptions')) {
-                    $this->view->stackTrace = $exception->getTraceAsString();
+                    $this->view->messages = $this->view->stackTraces = array();
+
+                    for (; $exception !== null; $exception = $exception->getPrevious()) {
```

Comment:
Please use a do-while loop here because it is the appropriate control structure to use. Your loop does of course work though. But when looking at this code, one may has to first think about what you're trying to achieve with it. With do-while it is obvious. Plus, the first check for `$exception !== null` is superfluous.